skymp5-server/cpp/server_guest_lib/FileDatabase.cpp

```diff
@@ -21,17 +21,24 @@ FileDatabase::FileDatabase(std::string directory_,
 size_t FileDatabase::Upsert(const std::vector<MpChangeForm>& changeForms)
 {
   auto p = pImpl->changeFormsDirectory;
+  size_t nUpserted = 0;
 
   for (auto& changeForm : changeForms) {
     auto filePath = p / changeForm.formDesc.ToString('_');
     std::ofstream f(filePath);
+    if (f) {
```

Comment:
failbit is set if error while opening occured, so this code should be correct

https://en.cppreference.com/w/cpp/io/basic_ofstream/open